app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -151,15 +155,24 @@ const SurveyAnimals: React.FC = () => {
     }
   };
 
-  const handleTelemetrySave = async (survey_critter_id: number, data: IAnimalTelemetryDevice[]) => {
+  const handleTelemetrySave = async (survey_critter_id: number, data: IAnimalTelemetryDeviceFile[]) => {
```

Comment:
This function is a bit complicated, could we break it down into more smaller functions? Maybe the ADD content is its own function, same with the EDIT?

Reply:
Also, adding comments inline is always okay. 

If we look at this or any code from the perspective of a brand new developer with no historical context on the project, are there any parts of the code that could use an explanation because what its doing, or why, is not obvious or involves some business knowledge that they just wouldn't have.